src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -898,6 +943,23 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
         cmd_helm_upgrade.extend(["--kubeconfig", kube_config])
     if kube_context:
         cmd_helm_upgrade.extend(["--kube-context", kube_context])
+    if container_registry_repository:
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.repository={}".format(container_registry_repository)])
+        cmd_helm_upgrade.extend(["--set", "global.isCustomRegistryEnabled={}".format(True)])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.releaseName={}".format("azurearck8s")])
+        if container_registry_username is not None and container_registry_password is not None:
+            cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.username={}".format(container_registry_username)])
+            cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.password={}".format(container_registry_password)])
+    if anonymous_pull:
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.username={}".format("")])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.image.password={}".format("")])
+
+    if default_registry:
```

Comment:
when switching from pcr to mcr, should we also enable auto upgrade of agents? We should think abt this

Reply:
The auto-upgrade will remain false.